src/playlist-controller.js

```diff
@@ -232,6 +235,14 @@ export class PlaylistController extends videojs.EventTarget {
     // we don't have to handle sourceclose since dispose will handle termination of
     // everything, and the MediaSource should not be detached without a proper disposal
 
+    if (this.hasManagedMediaSource_) {
+      this.handleStartStreaming_ = this.handleStartStreaming_.bind(this);
+      this.handleEndStreaming_ = this.handleEndStreaming_.bind(this);
+
+      this.mediaSource.addEventListener('startstreaming', this.handleStartStreaming_);
+      this.mediaSource.addEventListener('endstreaming', this.handleEndStreaming_);
```

Comment:
I guess we can remove the `hasManagedMediaSource_ ` flag and simply add these listeners alongside other listeners; we should not care about the type of the media source